From c207c353cbf883befa6070ccd80c2b4980a434fe Mon Sep 17 00:00:00 2001
From: Robert Church <robert.church@windriver.com>
Date: Thu, 13 Apr 2017 20:31:21 -0500
Subject: [PATCH 4/5] US92424: Add OSD support for persistent naming

This allows the manifest to provide udev generated /dev/disk/by-* links
to configure the OSDs without requiring any additional changes. The
'readlink -f' will produce the currently enumerated device node
associated with udev link.
---
 manifests/osd.pp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/manifests/osd.pp b/manifests/osd.pp
index 1972caa..2bda784 100644
--- a/manifests/osd.pp
+++ b/manifests/osd.pp
@@ -84,7 +84,7 @@ define ceph::osd (
 
     include ceph::params
 
-    $data = $name
+    $data = generate('/bin/bash','-c',"/bin/readlink -f ${name}")
 
     if $cluster {
       $cluster_name = $cluster
-- 
2.30.2

